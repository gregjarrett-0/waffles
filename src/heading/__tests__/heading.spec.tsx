import { render } from '@testing-library/react';

import { Heading } from '../index';

const sizes = ['xxlarge', 'xlarge', 'large', 'medium'] as const;

describe('Heading', () => {
  it('renders a h2 element by default containing the text', () => {
    const { container } = render(
      <Heading>Remarkable Taylor Swift live performance</Heading>,
    );

    const heading = container.querySelector('h2');

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(
      'Remarkable Taylor Swift live performance',
    );
  });

  it('sets the data attribute on the heading', () => {
    const { getByTestId } = render(
      <Heading data-testid="custom-heading">
        Remarkable Taylor Swift live performance
      </Heading>,
    );

    const heading = getByTestId('custom-heading');

    expect(heading).toBeInTheDocument();
  });

  it('default heading level associated with size could be overwrited', () => {
    const { container } = render(
      <Heading size="xxlarge" as="h6">
        Remarkable Taylor Swift live performance
      </Heading>,
    );

    const heading = container.querySelector('h6');

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(
      'Remarkable Taylor Swift live performance',
    );
  });

  describe('renders snapshot of', () => {
    sizes.forEach((size) => {
      it(`size ${size}`, () => {
        const { getByText } = render(
          <Heading key={size} size={size}>
            Test
          </Heading>,
        );

        const heading = getByText('Test');
        expect(heading).toMatchSnapshot();
      });
    });
  });

  describe('renders snapshot of inverted', () => {
    sizes.forEach((size) => {
      it(`size ${size}`, () => {
        const { getByText } = render(
          <Heading key={size} size={size} inverted>
            Test
          </Heading>,
        );

        const heading = getByText('Test');
        expect(heading).toMatchSnapshot();
      });
    });
  });
});
