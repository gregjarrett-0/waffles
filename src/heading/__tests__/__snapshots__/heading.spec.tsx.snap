// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders snapshot of inverted size large 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 16px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h3
  class="emotion-0"
>
  Test
</h3>
`;

exports[`Heading renders snapshot of inverted size medium 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 14px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h4
  class="emotion-0"
>
  Test
</h4>
`;

exports[`Heading renders snapshot of inverted size xlarge 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h2
  class="emotion-0"
>
  Test
</h2>
`;

exports[`Heading renders snapshot of inverted size xxlarge 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 20px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h1
  class="emotion-0"
>
  Test
</h1>
`;

exports[`Heading renders snapshot of size large 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 16px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h3
  class="emotion-0"
>
  Test
</h3>
`;

exports[`Heading renders snapshot of size medium 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 14px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h4
  class="emotion-0"
>
  Test
</h4>
`;

exports[`Heading renders snapshot of size xlarge 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h2
  class="emotion-0"
>
  Test
</h2>
`;

exports[`Heading renders snapshot of size xxlarge 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 20px;
  margin-bottom: 8px;
  line-height: 1.25;
}

<h1
  class="emotion-0"
>
  Test
</h1>
`;
